katas/content/superdense_coding/alice_sends_message/index.md

```diff
@@ -0,0 +1,18 @@
+**Inputs**:
+
+1. `qAlice` : Alice's part of the entangled pair of qubits.
+2. `message`: Two classical bits represented by a tuple of of two `Bool` variables to represent `bit1` and `bit2` respectively.
+
+**Goal**:
+Encode the message (two classical bits) by manipulating Alice's qubit.
+
+<details>
+  <summary><b>Need a hint?</b></summary>
+Manipulate Alice's half of the entangled pair to change the joint state of the two qubits to one of the following four states based on the value of message:
+
+- `(0, 0)`: $\ket{\Phi^{+}} = \frac{1}{\sqrt{2}} (\ket{00} + \ket{11})$
+- `(0, 1)`: $\ket{\Psi^{+}} = \frac{1}{\sqrt{2}} (\ket{01} + \ket{10})$
+- `(1, 0)`: $\ket{\Phi^{-}} = \frac{1}{\sqrt{2}} (\ket{00} - \ket{11})$
+- `(1, 1)`: $\ket{\Psi^{-}} = \frac{1}{\sqrt{2}} (\ket{01} - \ket{10})$
+
+</details>
```

Comment:
I just realized that we don't actually explain here how we test the solution. In all the previous katas, if the outcome of a task was expected to be a state, like here, the results would be a state comparison, but here for some reason we're talking about transfer. (And this was a problem in the classic kata too, I just never looked at it with a fresh eye! Revisiting everything as part of this migration is not a bad thing :-))

I suggest we rephrase a bit, to give more context on testing and to move expected encoding from hint to main task - strictly speaking, it's possible to implement superdense coding with a different mapping of messages to Bell states, it's just the customary approach:
```suggestion
Superdense coding protocol changes the joint state of the two qubits to one of the following four states based on the value of message:

- `(0, 0)`: $\ket{\Phi^{+}} = \frac{1}{\sqrt{2}} (\ket{00} + \ket{11})$
- `(0, 1)`: $\ket{\Psi^{+}} = \frac{1}{\sqrt{2}} (\ket{01} + \ket{10})$
- `(1, 0)`: $\ket{\Phi^{-}} = \frac{1}{\sqrt{2}} (\ket{00} - \ket{11})$
- `(1, 1)`: $\ket{\Psi^{-}} = \frac{1}{\sqrt{2}} (\ket{01} - \ket{10})$

Note that your solution is tested as part of an end-to-end implementation of the protocol; the goal is to get the message transmitted correctly.
```

Reply:
Nice catch, Done :)